aea/aea.py

```diff
@@ -223,7 +229,35 @@ def _handle(self, envelope: Envelope) -> None:
             return
 
         for handler in handlers:
-            handler.handle(msg)
+            self._handle_message_with_handler(msg, handler)
+
+    def _handle_message_with_handler(self, message: Message, handler: Handler) -> None:
+        """Handle one message with one predefined handler.
+
+        :param message: message to be handled.
+        :param handler: handler suitable for this message protocol.
+        """
+        with ExecTimeoutThreadGuard(self._execution_timeout) as timeout_result:
+            handler.handle(message)
+
+        if timeout_result.is_cancelled_by_timeout():
+            logger.warning(
+                f"Handler `{handler}` was terminated by timeout == {self._execution_timeout}"
```

Comment:
```suggestion
                "Handler `{}` was terminated with timeout {} seconds".format(handler, self._execution_timeout)
```
For consistency.

Reply:
done. but curious why not f string?  it's supported from python3.6, in this case, is suitable for text formating

Reply:
Just a convention atm. Can discuss.